### PR TITLE
Add spec dev endpoints  /spec/action/:spec-ns/:spec-id (Connect  #1876)

### PR DESCRIPTION
**Pull request by tangrammer**
- [ ] **Update release notes if necessary**

- [x] Fix edn files dependency

```
dev env use test.edn

and test env doesn't use dev.edn config
``` 

- [x] Use upload files instead of local-server in endpoints-test

- [x] Add dev endpoint: GET /spec/describe/:spec-ns transit-format 
Example:
http://t1.lumen.local:3000/spec/describe/:akvo.lumen.lib.aggregation/dataset

- [x] Add dev endpoint: GET /spec/sample/:spec-ns  
Example: http://t1.lumen.local:3000/spec/sample/:akvo.lumen.lib.aggregation/dataset
- [x] Improve middleware composition using `:akvo.lumen.component.handler/common-middleware`
- [x] Add dev endpoint: POST /spec/conform/:spec-ns

### CODE REVIEW

**Comment by tangrammer** [comment]
@kardan ready to review!

File: backend/dev/src/dev/endpoints.clj [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            [ring.util.response :refer [response]]
+            [integrant.core :as ig]))
+;; working in adapt https://github.com/akvo/akvo-lumen/blob/310bd7cbc3221bf889ad592f2d9e91a572f06c00/backend/dev/src/akvo/lumen/local_server.clj
+;; use edn middleware instead of json middleware
+(defn routes [opts]
+  ["/:spec-ns/:spec-id" ["" {:get {:parameters {:path-params {:spec-ns string?
+                                                              :spec-id string?}}
+                                   :handler (fn [{{:keys [spec-ns spec-id]} :path-params}]
+                                              (let [spec (keyword (str (apply str (next (seq spec-ns))) "/" spec-id))]
+                                                (response {:namespace spec-ns
```
Comment (kardan) [inline backend/dev/src/dev/endpoints.clj]: A bit odd formatting here

File: backend/dev/resources/dev.edn [hunk 2]
```diff
@@ -1,60 +1,22 @@
...
-                               :piwik-site-id "165"}
-
+ :akvo.lumen.test-utils/wrap-jwt-mock {:keycloak #ig/ref :akvo.lumen.component.keycloak/data}
  :akvo.lumen.migrate/migrate {:seed
                               {:tenants
                                [{:db_uri "jdbc:postgresql://postgres/lumen_tenant_1?user=lumen&password=password&ssl=true"
-                                 :label "t1"
-                                 :title "Tenant 1"
-                                 :plan "standard"}
+                                 :label  "t1"
```
Comment (kardan) [inline backend/dev/resources/dev.edn]: Whitespace is only whitespace - but

File: backend/test/akvo/lumen/endpoints_test/commons.clj [hunk 3]
```diff
@@ -79,3 +85,114 @@
...
+  (cond->
+      {:request-method method
+       :uri uri
+       :headers {"host" "t1.lumen.local:3030" "content-type" "application/json"}
+       :path-info uri
+       :server-port 3030,
+       :server-name "t1.lumen.local",
+       :remote-addr "localhost",
+       :scheme :http,
+       ;;       :body-params body
```
Comment (kardan) [inline backend/test/akvo/lumen/endpoints_test/commons.clj]: Remove?

File: backend/dev/resources/dev.edn [hunk 4]
```diff
@@ -1,60 +1,22 @@
...
+ :akvo.lumen.component.handler/handler {:endpoints [["/local-development" [#ig/ref :akvo.lumen.utils.local-server/endpoint]]
+                                                    #ig/ref :dev.endpoints/spec-api]}
 
  :akvo.lumen.migrate/migrate {:seed
                               {:tenants
                                [{:db_uri "jdbc:postgresql://postgres/lumen_tenant_1?user=lumen&password=password&ssl=true"
-                                 :label "t1"
-                                 :title "Tenant 1"
-                                 :plan "standard"}
+                                 :label  "t1"
```
Comment (kardan) [inline backend/dev/resources/dev.edn]: Whitespace is only whitespace, but no need to be greedy.

**Comment by tangrammer** [comment]
@kardan Added changes to the review 👍

**Pull request merged**